Objective-C/Tests/ConcurrentTest.m

```diff
@@ -274,9 +274,9 @@ - (void) testConcurrentPurgeDocs {
     
     [self concurrentRuns: kNConcurrents waitUntilDone: YES withBlock: ^(NSUInteger rIndex) {
         for (CBLDocument* doc in docs) {
-            NSError* error;
-            if (![self.db purgeDocument: doc error: &error])
-                AssertEqual(error.code, 404);
+            [self mayHaveException: NSInvalidArgumentException in: ^{
```

Comment:
It seems wrong that this can throw an exception — that indicates a fatal programmer mistake, not a runtime error. Where does that exception come from? Can we change it to an NSError?

Reply:
Agreed. I have been back and forth on this. I will change to return NOT-FOUND error. @borrrden probably will review this again for the error review task.